godot-core/src/builtin/basis.rs

```diff
@@ -0,0 +1,980 @@
+            "got {got}, expected {expected}",
+            got = Basis::from_axis_angle(Vector3::FORWARD, PI + FRAC_PI_2) * Vector3::RIGHT,
+            expected = Vector3::UP
+        );
+    }
+
+    // Translated from Godot
+    #[test]
+    fn basis_euler_conversions() {
+        let euler_order_to_test: Vec<EulerOrder> = vec![
+            EulerOrder::XYZ,
+            EulerOrder::XZY,
+            EulerOrder::YZX,
+            EulerOrder::YXZ,
+            EulerOrder::ZXY,
+            EulerOrder::ZYX,
+        ];
+
+        let vectors_to_test: Vec<Vector3> = vec![
+            Vector3::new(0.0, 0.0, 0.0),
+            Vector3::new(0.5, 0.5, 0.5),
+            Vector3::new(-0.5, -0.5, -0.5),
+            Vector3::new(40.0, 40.0, 40.0),
+            Vector3::new(-40.0, -40.0, -40.0),
+            Vector3::new(0.0, 0.0, -90.0),
+            Vector3::new(0.0, -90.0, 0.0),
+            Vector3::new(-90.0, 0.0, 0.0),
+            Vector3::new(0.0, 0.0, 90.0),
+            Vector3::new(0.0, 90.0, 0.0),
+            Vector3::new(90.0, 0.0, 0.0),
+            Vector3::new(0.0, 0.0, -30.0),
+            Vector3::new(0.0, -30.0, 0.0),
+            Vector3::new(-30.0, 0.0, 0.0),
+            Vector3::new(0.0, 0.0, 30.0),
+            Vector3::new(0.0, 30.0, 0.0),
+            Vector3::new(30.0, 0.0, 0.0),
+            Vector3::new(0.5, 50.0, 20.0),
+            Vector3::new(-0.5, -50.0, -20.0),
+            Vector3::new(0.5, 0.0, 90.0),
+            Vector3::new(0.5, 0.0, -90.0),
+            Vector3::new(360.0, 360.0, 360.0),
+            Vector3::new(-360.0, -360.0, -360.0),
+            Vector3::new(-90.0, 60.0, -90.0),
+            Vector3::new(90.0, 60.0, -90.0),
+            Vector3::new(90.0, -60.0, -90.0),
+            Vector3::new(-90.0, -60.0, -90.0),
+            Vector3::new(-90.0, 60.0, 90.0),
+            Vector3::new(90.0, 60.0, 90.0),
+            Vector3::new(90.0, -60.0, 90.0),
+            Vector3::new(-90.0, -60.0, 90.0),
+            Vector3::new(60.0, 90.0, -40.0),
+            Vector3::new(60.0, -90.0, -40.0),
+            Vector3::new(-60.0, -90.0, -40.0),
+            Vector3::new(-60.0, 90.0, 40.0),
+            Vector3::new(60.0, 90.0, 40.0),
+            Vector3::new(60.0, -90.0, 40.0),
+            Vector3::new(-60.0, -90.0, 40.0),
+            Vector3::new(-90.0, 90.0, -90.0),
+            Vector3::new(90.0, 90.0, -90.0),
+            Vector3::new(90.0, -90.0, -90.0),
+            Vector3::new(-90.0, -90.0, -90.0),
+            Vector3::new(-90.0, 90.0, 90.0),
+            Vector3::new(90.0, 90.0, 90.0),
+            Vector3::new(90.0, -90.0, 90.0),
+            Vector3::new(20.0, 150.0, 30.0),
+            Vector3::new(20.0, -150.0, 30.0),
+            Vector3::new(-120.0, -150.0, 30.0),
+            Vector3::new(-120.0, -150.0, -130.0),
+            Vector3::new(120.0, -150.0, -130.0),
+            Vector3::new(120.0, 150.0, -130.0),
+            Vector3::new(120.0, 150.0, 130.0),
+        ];
```

Comment:
This constant is used in `itest` as well, unfortunately there's currently no way to reuse it, without adding it to the user-facing crate. Maybe in the future I could add a cargo feature for this.